### DISCUSSION TITLE
Can you get parameters from a querystring?

### DISCUSSION THREAD

**Discussion by frostymountain** [inkle/ink#864] [Writing in ink]
Is there a way to read parameters from a querystring when your ink game is opened in a browser?

Alternatively, is there a way to run a block of javascript that will set an ink variable to a parameter from a querystring?

**Comment by Selsynn** [comment]
Hey, 

I suppose you will be using the web export, else your question is not really making any sense. 
In the web exported folder, you can see that there is main.js file. It's a javascript file and you are free to do whatever you want with it.

So you can read the querystring parameter (I think it's pure vanilla js) and do whatever treatment you want. 
If you update the ink files later, think about exporting only the ink data, and not regenerate all the web export (else you will loose your changes)

If you have more technical questions on how to do stuff in the main.js, I strongly encourage you to post it in the inkjs project https://github.com/y-lohse/inkjs

**Comment by frostymountain** [comment]
Yes, I am using the web export.

So, quick question, if I had a line to main.js and put a querystring parameter into a const, how do I get that value into a game variable, so I 
can show it to a player?

**Reply by Selsynn** [reply]
put it into a ink variable like this `_inkStory.variablesState.$('your_ink_var', queryParam);`

(Source https://github.com/y-lohse/inkjs#differences-with-the-c-api )

**Comment by frostymountain** [comment]
I think I came up with a shortcut and don't need to go this way. Thanks.
